File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaPositionApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaPositionApi {

    @GET("/api/v3/position/current-position")
    Call<ResponseResult> currentPosition(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/position/history-position")
    Call<ResponseResult> historyPosition(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/position/adlRank")
    Call<ResponseResult> adlRank();
}

File: bitget-node-sdk-api/src/lib/v3/UtaPositionApi.ts
import {BaseApi} from '../BaseApi';

export class UtaPositionApi extends BaseApi {

    // GET /api/v3/position/current-position
    currentPosition(qsOrBody: object) {
        const url = '/api/v3/position/current-position';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/position/history-position
    historyPosition(qsOrBody: object) {
        const url = '/api/v3/position/history-position';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/position/adlRank
    adlRank(qsOrBody: object) {
        const url = '/api/v3/position/adlRank';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}
